addons/sale_project/views/project_task_views.xml

```diff
@@ -78,8 +78,8 @@
             </xpath>
             <xpath expr="//field[@name='partner_phone']" position="after">
                 <field name="sale_order_id" attrs="{'invisible': True}" groups="sales_team.group_sale_salesman"/>
-                <field name="sale_line_id" groups="!sales_team.group_sale_salesman" string="Sales Order Item" options='{"no_open": True}' readonly="1" context="{'create': False, 'edit': False, 'delete': False}"/>
-                <field name="sale_line_id" groups="sales_team.group_sale_salesman" string="Sales Order Item" options='{"no_create": True}' readonly="0" context="{'create': False, 'edit': False, 'delete': False, 'with_price_unit': True}"/>
+                <field name="sale_line_id" groups="!sales_team.group_sale_salesman" attrs="{'invisible': [('partner_id', '=', False)]}" string="Sales Order Item" options='{"no_open": True}' readonly="1" context="{'create': False, 'edit': False, 'delete': False}"/>
+                <field name="sale_line_id" groups="sales_team.group_sale_salesman" attrs="{'invisible': [('partner_id', '=', False)]}" string="Sales Order Item" options='{"no_create": True}' readonly="0" context="{'create': False, 'edit': False, 'delete': False, 'with_price_unit': True}"/>
```

Comment:
You will need to rebase for this, they also need to be invisible when allow_billable = False and a fix was recently merged for that